js/battle.js

```diff
@@ -2215,6 +2215,8 @@ var Battle = (function () {
 		this.logElem = this.logFrameElem.children();
 		this.logFrameElem.append('<div class="inner-preempt"></div>');
 		this.logPreemptElem = this.logFrameElem.children().last();
+		this.logFrameElem.append('<div class="battle-options"></div>');
+		this.optionsElem = this.logFrameElem.children().last().html('<button name="toggleIgnoreSpects">' + (this.ignoreSpects ? 'Unignore' : 'Ignore') + ' Spectators</button>');
```

Comment:
This second line should be moved to `client-battle.js`. It's not like it does anything in replays.